src/Umbraco.Web/Trees/TreeAttribute.cs

```diff
@@ -1,65 +1,87 @@
 ﻿using System;
-using Umbraco.Web.Models.Trees;
 
 namespace Umbraco.Web.Trees
 {
+    // fixme move
+    [Flags]
+    public enum TreeUse
+    {
+        None = 0,
+        Main = 1,
+        Dialog = 2,
+    }
+
+    public static class TreeUseExtensions
+    {
+        public static bool Has(this TreeUse use, TreeUse uses)
```

Comment:
IMO, this is bad naming. From the name I would expect 
```
var treeUse = TreeUse.Main | TreeUse.Dialog;
var expectTrue = treeUse.Has(TreeUse.Dialog); //This is false
```

Not sure this method is even used.

Reply:
The implementation now does what I think is correct if the name is `Has`, but i my example this would return `true`. 

But if you want it to be `true` only if all flags are set, then the implementation needs to be 
`(use | uses) == uses`, but then I again think the naming is bad.

Reply:
Have refactored and pushed (see commit d24b7f2aba) to cleanup code. Some clarifications:
* Enums have a builtin `value.HasFlag(Enum flag)` method that determines whether all bits from `flag` are set in `value`, though other bits can be set in `value` too.
* That method is annoying because its parameter is of type `Enum`, not the actual type of any enum, so Intellisense does not kick.
* The `Has` method was meant to provide a typed version of `HasFlag` - behaving exactly in the same way - returning `true` if all flags are set (but not necessarily only those flags) - but with better Intellisense
* We don't need a method to determine if "only all flags are set", that's an `==` comparison

And so to try and make things clearer, I've renamed `Has` to `HasFlagAll` and then we also have `HasFlagAny` - both complementing the original `HasFlag` with better Intellisense support. And there's unit tests for the three of them.

Is this making more sense?

Reply:
The new names and the generic implementation are better :)